### ISSUE TITLE
Vertex color from textools are giving the wrong values.

### ISSUE THREAD

**Issue by Leonature26** [franMarz/TexTools-Blender#218]
I just realized that the red values that I'm getting from textools are all wrong. When I inputted a red value of 0.125 it actually applies 0.389 for some reason. I only noticed this when I used blender's native tools and the colors aren't matching up. I need a specific vertex color value so this is gonna be a pain to correct since I've done multiple assets already.
![image](https://github.com/user-attachments/assets/7be7c406-1ca6-49a7-aa27-9611853701ef)

**Comment by Leonature26** [comment]
I'm not sure how they can be so wildly different. Am I missing something in the settings? I've tried it in maya too and the textool values are indeed the wrong one.
![image](https://github.com/user-attachments/assets/5ac5a2a9-771b-4520-b52f-b5d1c4d23ec1)

**Comment by franMarz** [comment]
The original author set a gamma correction that transforms your chosen color under Color ID before applying it. This has been like this for years, maybe you're correct and it has no sense, but no one has had a problem with this before, so no one fixed it.
I can make that gamma correction to be an option in the addon preferences, disabled by default. Or maybe it was a good idea for the Material Mode, and not as good for the Vertex Colors Mode. In that case, I'll make the transform to only run in the Material Mode.
For now, I think I'll go for the first option, make the correction optional for both Color ID modes.
If you can isolate the objects affected by this, you can setup a script to get the vertex color per loop and reassign it with an inverse transform per color channel ( color[i] = pow(color[i], 2.2) ). I can help you with that later.

**Comment by Leonature26** [comment]
I've already fixed the errors by using a script to select by vertex color then just reapply a correct value using blender's native tool. 

Having that option to disable the gamma corrrection would be really helpful.

**Comment by franMarz** [comment]
TL;DR: I'll made the correction optional for the Color ID Vertex Color mode.

I've been looking into this. If you compare how Blender manage color between the Vertex Paint mode and the Shader Editor, for example, you'll see the same differences when it comes to color pickers. The internal gamma correction is different when presented to the user through the color pickers between modes that operate in color data. It depends on whether that data directly feeds a render engine alongside externally, user supplied textures, needing an inverse gamme to be applied internally before going into the renderer to be in line with those textures (Material Editor), or is internal raw color data that doesn't need that, so you see basically internal linear color values in the channels of the color picker for the same colors, with the gamma applied (Vertex Paint mode). A bit confusing, but that differences you show in your second post comes from Blender, not TexTools.

The color transform in the Color ID Vertex Color Mode was introduced by me a long time ago to get the same colors that in the Color ID Material Mode, despite the channe values being different, just for the that confusing behaviour. But I get know that it's better to translate the data unmodified by default in the Vertx Color mode as it should remain as raw data for a good internal behaviour, despite how different it looks in the 3D View compared to materials with a color applied that shares the sames values per color channel.
